webapp/src/components/sidebar_buttons/index.js

```diff
@@ -1,21 +1,20 @@
 import {connect} from 'react-redux';
 import {bindActionCreators} from 'redux';
 
-import {getReviews, getUnreads, getYourPrs, getYourAssignments} from '../../actions';
+import {getReviews, getUnreads, getYourPrs, getYourAssignments, updateRhsState} from '../../actions';
 
 import SidebarButtons from './sidebar_buttons.jsx';
 
 function mapStateToProps(state) {
     return {
         connected: state['plugins-github'].connected,
-        username: state['plugins-github'].username,
         clientId: state['plugins-github'].clientId,
         reviews: state['plugins-github'].reviews,
         yourPrs: state['plugins-github'].yourPrs,
         yourAssignments: state['plugins-github'].yourAssignments,
         unreads: state['plugins-github'].unreads,
         enterpriseURL: state['plugins-github'].enterpriseURL,
-        org: state['plugins-github'].organization,
+        showRHSPlugin: state['plugins-github'].rhsPlugAction,
```

Comment:
I agree it's not a common case. The scenario is we have an action that is stored in the Redux state, in the mapDispatchToProps(dispatch) you do not have access to the store so you don't have access to the action to bind it with the dispatch. 

In that case, we went with the approach of binding it before storing in the state and then getting it from the state as a prop and calling the function directly. 

@lieut-data This is the "unsual" workflow that I mentioned. Would there be any value to document this as a plugin "Common Workflows" ? @crspeller Thoughts?

Reply:
ahhh gotcha, because it's a dynamic action that's stored in the redux store.

I think there is another parameter that's passed into the `mapDispatchToProps` func that has the connected component's props available to it. So you could do something like:
```
function mapDispatchToProps(dispatch, ownProps) {
    return {
        actions: bindActionCreators({
            getReviews,
            getUnreads,
            getYourPrs,
            getYourAssignments,
            updateRhsState,
            showRHSPlugin: ownProps['plugins-github'].rhsPlugAction,
        }, dispatch),
    };
}
```
https://react-redux.js.org/using-react-redux/connect-mapdispatch#arguments
(not 100% sure if it'll work but worth a try)

The reason i'm recommending `showRHSPlugin` to be part of the `actions` prop is to remove the awkwardness from the client component so it doesn't have to do `this.props.showRHSPlugin()` in one case and for all other actions do `this.props.actions.ACTION1()`

Reply:
Yes, I explored the ownProps route where the showPluginAction could be passed as a prop to the SidebarButtons component so it would be available there in ownProps. 
However, that component is a child component of SidebarHeader that is [registered in the webApp here](https://github.com/marianunez/mattermost-plugin-github/blob/9a7daf919faf7bd648496d9546e8a743e5567764/webapp/src/index.js#L21) which receives the component the webapp will include in the corresponding "pluggable" section. This accepts a type of component (where the props can't be set) or a JSX functional component. The SidebarButtons is heavily using the state so couldn't easily change to a stateless functional component. 

> The reason i'm recommending showRHSPlugin to be part of the actions prop is to remove the awkwardness from the client component so it doesn't have to do this.props.showRHSPlugin() in one case and for all other actions do this.props.actions.ACTION1()

I agree it hurts the eyes 😖 but was the best approach possible for this case.

Reply:
Spoke to @marianunez offline regarding this complicated case. Thinking a bit more about it, _maybe_ the following solution could work. i'm 1/5 for this as this way all awkwardness/funkiness stays in this one file and the client/presentation component uses that action as any other action. But I'll leave it upto @marianunez to make the final call 👍 

Suggestion to get it to work:
```
export default connect(mapStateToProps)(connect(null, mapDispatchToProps)(SidebarButtons));

```


EDIT: I was going to try out the change above, but @marianunez helped me test that change so looks like it is working.

Reply:
> Suggestion to get it to work: ``export default connect(mapStateToProps)(connect(null, mapDispatchToProps)(SidebarButtons));``

This did work as another possible implementation to being able to access the action stored in the state in the newProps in the mapDispatchToProps. 

However, using the newProps approach does reduce the React optimization on components re-rendering (see [React doc here](https://react-redux.js.org/using-react-redux/connect-mapdispatch#arguments) ). Given that the only benefit of this approach would be to call the action in the component as:

``this.props.action.showRHSPlugin()``  vs 
``this.props.showRHSPlugin()`` 

then it doesn't seem much of  a gain to lose performance over. 

Thanks for the review and researching other possible options @ali-farooq0

Reply:
Thanks @marianunez!